### DISCUSSION TITLE
Github Webhooks getting sent without X-Hub-Signature headers

### DISCUSSION THREAD

**Discussion by yonirab** [community/community#24753] [Apps, API and Webhooks]
<p>Our organization has many private repos.<br>
In the past, when we created new Github Webhooks from the Github UI, the webhooks seem to have been created with a “Secret” which caused the webhook payload headers to include <code>X-Hub-Signature</code> and <code>X-Hub-Signature-256</code>.<br>
For some reason, new webhooks seem to get created without a “secret”, and those headers seem to be missing.<br>
We use these webhooks to trigger CI builds on Jenkins, and the absence of the headers seems to cause a problem for Jenkins.<br>
How do I get newly created webhooks to have a “secret” and to include the headers?</p>

**Comment by canuckjacq** [comment, marked as answer]
<p>:wave: Welcome!</p>
<p>In the create webhook UI, you should have a field to enter <a href="https://docs.github.com/en/developers/webhooks-and-events/webhooks/securing-your-webhooks#setting-your-secret-token">your secret token</a>. When you have done that, you should definitely be receiving the signature headers.</p>

**Comment by donation93** [comment]
<p>Right , should to do that thabks</p>

**Comment by yonirab** [comment]
<p>Thank you so much! I now understand what I was missing.</p>
